### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by dkazanc**

### CODE REVIEW

File: docs/source/doc-conda-requirements.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+dependencies:
+  - python=3.9
+  - sphinx
+  - sphinx_rtd_theme
+  - nbsphinx
+  - pandoc
+  - jinja2
+  - sphinx-panels
+  - sphinx-copybutton
+  - tomopy
```
Comment (namannimmo10) [inline docs/source/doc-conda-requirements.yml:14]: why do we need tomopy here?

Comment (dkazanc) [reply docs/source/doc-conda-requirements.yml:14]: I wonder... If we get the templates locally then we don't really need it?

Comment (jessicavers) [reply docs/source/doc-conda-requirements.yml:14]: Yes, I think it doesn't need to be there anymore. I think I added it maybe when I was running the python file templates/tomopy/yaml_generator_tomopy.py inside the github action, then I forgot to remove it.

Comment (namannimmo10) [reply docs/source/doc-conda-requirements.yml:14]: yes, thanks Jessica, would be nice if you remove it - the size is huge.

File: README.rst [hunk 2]
```diff
@@ -1,51 +1,64 @@
...
 
 * Install the module as described in "Install as a Python module"
 * Execute the python module with :code:`httomo <args>`
 * For help with the command line interface, execute :code:`httomo --help`
 
+Am example of running the code with test data:
+==============================================
+
+* Go to the home directory and run: :code:`python -m httomo testdata/tomo_standard.nxs . cpu`
+
```
Comment (namannimmo10) [inline README.rst:51]:
```suggestion
```

File: docs/source/explanation/templates.rst [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+name of the method in that module. HTTomo will interpret this in Python
+by importing the method:
+
+.. code-block:: python
+
+    from tomopy.misc.corr import median_filter3d
+
+After the method's name follows the list of parameters for that method. Parameters :code:`data_in` 
+and :code:`data_out` define the names of the input and the output datasets respectively. Then :code:`size` 
+is a specific to the method parameter.
```
Comment (namannimmo10) [inline docs/source/explanation/templates.rst:28]:
```suggestion
is specific to the method parameter. 
```

File: docs/source/explanation/yaml.rst [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+Using a YAML file to specify a pipeline of functions for data processing is a common
+and practical approach to creating a reproducible data analysis workflow. This approach
+can help to ensure that your pipeline runs correctly and consistently over time,
+regardless of the platform or environment in which it is executed. By using a YAML file
+to define your pipeline, we are providing a user interface that is simple and intuitive for scientists
+to use. This can be especially helpful for those who are less familiar with programming in general
+or are new to the specific tools and libraries you are using.
+
+We have a YAML checker that can help you validate your YAML file, before you use it to run your pipeline,
+and we highly recommend that you use it. The checker will help you to identify any errors in your YAML file.
```
Comment (namannimmo10) [inline docs/source/explanation/yaml.rst:69]: would be nice to move this to the yaml checker section...

File: docs/source/howto/yaml_checker.rst [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+* If you pass :code:`IN_DATA` (path to the data) along with the yaml config, as:
+
+.. code-block:: console
+
+    $ python -m httomo check config.yaml IN_DATA
+
+That will check that the paths to the data and keys in the :code:`YAML_CONFIG` file match the paths and keys in the input file (:code:`IN_DATA`).
+
+If you have the following loader in your yaml file:
+.. code-block:: yaml
```
Comment (namannimmo10) [inline docs/source/howto/yaml_checker.rst:187]: Going haywire here!
```suggestion
If you have the following loader in your yaml file:

.. code-block:: yaml
```

File: docs/source/backends/list.rst [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+If the package has a modular structure with an easy access to every method, e.g., `TomoPy <https://tomopy.readthedocs.io>`_ software or `scikit <https://scikit-image.org/>`_ library, then the integration process is `straightforward <https://diamondlightsource.github.io/httomo/explanation/faq.html>`_.  
+More complicated in structure packages would need an additional wrapping, see for instance the reconstruction method in the `HTTomolib <https://github.com/DiamondLightSource/httomolib/blob/master/httomolib/recon/algorithm.py#L72>`_ library. 
+
+
+TomoPy software (CPU)
+---------------------------------
+`TomoPy <https://tomopy.readthedocs.io>`_ is an open-source Python package for tomographic data processing and image reconstruction developed at `The Advanced Photon Source <https://www.aps.anl.gov/>`_ in Illinois, USA. 
+Being active since 2013 it gained a `large audience <https://github.com/tomopy/tomopy>`_ of users and contributors in tomographic imaging community.
+
+* An open-source package in Python and C for data data processing and reconstruction.  With a few exceptions, TomoPy is mostly CPU processing library. In HTTomo we expose only CPU modules of TomoPy.
```
Comment (namannimmo10) [inline docs/source/backends/list.rst:16]:
```suggestion
* An open-source package in Python and C for data processing and reconstruction.  With a few exceptions, TomoPy is mostly CPU processing library. In HTTomo we expose only CPU modules of TomoPy. 
```

File: .github/workflows/doc_conda.yml [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+  pull_request:
+    branches:
+      - main
+      - documentation
+      - docs
+  push:
+    branches:
+      - main
+      - documentation
+      - docs
```
Comment (namannimmo10) [inline .github/workflows/doc_conda.yml:14]: since we are merging this to `main`, we don't have to run the workflow on these branches...

Comment (dkazanc) [reply .github/workflows/doc_conda.yml:14]: yep, please correct that?

**Comment by namannimmo10** [comment]
before you push anything here, please pull down the changes first, otherwise you'll undo other changes.

**Comment by namannimmo10** [comment]
Would you like to add a better theme to the docs? See

https://github.com/ionelmc/python-aspectlib/blob/master/docs/conf.py#L31-L32

and the corresponding docs: https://python-aspectlib.readthedocs.io/en/latest/examples.html#debugging

**Comment by yousefmoazzam** [comment]
> Would you like to add a better theme to the docs? See
> 
> https://github.com/ionelmc/python-aspectlib/blob/master/docs/conf.py#L31-L32
> 
> and the corresponding docs: https://python-aspectlib.readthedocs.io/en/latest/examples.html#debugging

I'm up for exploring other themes! :) However, I can't say I am fond of the proposed new one, I think I prefer the old one?... I think it is also currently breaking the docs build, the github action output isn't happy about the package the new theme should come from or something:
```
Updating 'httomo' env from conda env update...
  /usr/share/miniconda/condabin/conda env update --name httomo --file ./docs/source/doc-conda-requirements.yml
  Collecting package metadata (repodata.json): ...working... done
  Solving environment: ...working... failed
  Warning: 
  ResolvePackageNotFound: 
    - sphinx_py3doc_enhanced_theme
  
  
  
  ResolvePackageNotFound: 
    - sphinx_py3doc_enhanced_theme
  

Error: The process '/usr/share/miniconda/condabin/conda' failed with exit code 1
```

**Comment by namannimmo10** [comment]
Looks like the theme is available only on pypi. While there are other fixes to make it work, you can revert the commit if it is not to your taste.

File: docs/source/utilities/yaml_checker.rst [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+        standard_tomo:
+          name: tomo
+          data_path: entry1/tomo_entry/data/data
+          image_key_path: entry1/tomo_entry/instrument/detector/image_key
+          preview:
+            -
+            - start: 30
+              stop: 60
+            -
+        pad: 0
```
Comment (namannimmo10) [inline docs/source/utilities/yaml_checker.rst:107]:
```suggestion
          preview:
            -
            - start: 30
              stop: 60
            -
          pad: 0
```

**Pull request merged**